azurelinuxagent/common/protocol/util.py

```diff
@@ -317,34 +270,33 @@ def clear_protocol(self):
                 return
             logger.error("Failed to clear protocol endpoint: {0}", e)
 
-    def get_protocol(self, by_file=False):
+    def get_protocol(self):
         """
-        Detect protocol by endpoints, if by_file is True,
-        detect MetadataProtocol in priority.
+        Detect protocol by endpoint.
         :returns: protocol instance
         """
 
         if self._protocol is not None:
             return self._protocol
 
-        try:
-            self._protocol = self._get_protocol()
-            return self._protocol
-        except ProtocolNotFoundError:
-            pass
-        logger.info("Detect protocol endpoints")
-        protocols = [prots.WireProtocol]
-
-        if by_file:
-            tag_file_path = self._get_tag_file_path()
-            if os.path.isfile(tag_file_path):
-                protocols.insert(0, prots.MetadataProtocol)
-        else:
-            protocols.append(prots.MetadataProtocol)
-        protocol_name, protocol = self._detect_protocol(protocols)
+        protocol_file_path = self._get_protocol_file_path()
+        if os.path.isfile(protocol_file_path):
+            protocol_name = fileutil.read_file(protocol_file_path)
+            if protocol_name == prots.WireProtocol:
+                endpoint = self.get_wireserver_endpoint()
+                self._protocol = WireProtocol(endpoint)
+                return self._protocol
+            elif protocol_name == prots.MetadataProtocol:
```

Comment:
Checking Protocol file for MetadataServer is not sufficient for the manual upgrade case where we upgrade from MetadataServer to WireServer by restarting the agent (WireServer is written to protocol file so we never enter cleanup branch).